### PR TITLE
[SA-46] Add ARCHITECTURE doc

### PR DESCRIPTION
**Pull request by ksclarke**
* Add GitHub Actions
* Add ARCHITECURE.md doc
* Fix test so scratch space directory is created if it doesn't already exist
* Rename the OpenAPI spec file in preparation for having an AsyncAPI YAML file in the future
* Swap out unspecified host address with localhost address for tests

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+| Handler(s) | These handle incoming HTTP requests (e.g. application status requests) | [src/main/java/edu/ucla/library/avpairtree/handlers](https://github.com/UCLALibrary/av-pairtree/tree/main/src/main/java/edu/ucla/library/avpairtree/handlers) |
+| CsvItem | This is an object which represents a single item (or row) from the CSV file | [src/main/java/edu/ucla/library/avpairtree/CsvItem.java](https://github.com/UCLALibrary/av-pairtree/blob/main/src/main/java/edu/ucla/library/avpairtree/CsvItem.java) |
+| CsvItemCodec | This codec implements a JSON (de)serialization of CsvItem so that it can be sent over the event bus | [src/main/java/edu/ucla/library/avpairtree/CsvItemCodec.java](https://github.com/UCLALibrary/av-pairtree/blob/main/src/main/java/edu/ucla/library/avpairtree/CsvItemCodec.java) |
+
+Actions (e.g., the parsing CSV files, conversion of media files, or storage of media files in a Pairtree structure, etc.) are performed by the application's various verticles (e.g., WatcherVerticle, ConverterVerticle, PairtreeVerticle, etc.) Cf. the `verticles` directory for examples.
```
Comment (DRickard) [inline ARCHITECTURE.md:36]: Minor type--missing "of" after "parsing"

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+After all the A/V files in a CSV file have been processed, the input CSV is updated to include the resources' new access URLs (i.e. the URLs of the media files as served by the media server) and written back out to the file system.
+
+![Overview diagram for av-pairtree's components](docs/images/av_pairtree_overview.svg)
+
+## Expected CSV Structure
+
+A CSV file that is going to be processed by av-pairtree should have two required columns: `File Name` and `ItemARK`. The first is used to retrieve the media file to be processed and the second is used to create the Pairtree structure. If the file has been previously processed (or has been processed by the Bucketeer application), it will also have a `IIIF Access URL` column. That's fine. However, older CSV files may have 'iiif_access_url` as a column header. Any CSVs with that column header should be manually updated before processing with av-pairtree.
```
Comment (markmatney) [inline ARCHITECTURE.md:17]: There's an apostrophe in front of `iiif_access_url` but should be a grave quote

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+| audio.bit.rate | The bit rate used to encoding the audio stream | 128000 |
+| audio.channels | The number of channels in the audio stream | 2 |
+| iiif.access.url | The URL pattern into which to insert the Pairtree path | N/A |
+| conversion.workers | The number of cores to use for audio file conversion | 2 |
+
+## Documentation
+
+This project has a `README.md` file and an `ARCHITECTURE.md` file (what you're currently reading) in its root directory.
+
+The application also uses OpenAPI and AsyncAPI files to define its public and internal messaging APIs. These can be found in the [`src/main/resources`](https://github.com/UCLALibrary/av-pairtree/tree/main/src/main/resources) directory. The files are `av-pairtree-openapi.yaml` and `av-pairtree-asyncapi.yaml`.
```
Comment (markmatney) [inline ARCHITECTURE.md:74]: Just flagging that there isn't an AsyncAPI doc yet, but I assume it may be added soon?

**Review by markmatney — CHANGES_REQUESTED** [review]
One minor Markdown syntax error (but which is necessary for proper syntax highlighting, at least in GitHub's diff viewer)

File: ARCHITECTURE.md (hunk 3 above)
Comment (ksclarke) [reply ARCHITECTURE.md:74]: Yes, the file name is just in anticipation of there needing to be another yaml file in that same dir with basically the same name.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Pull request merged**